### PR TITLE
TextareaControl: Update styles

### PR DESCRIPTION
**Pull request by mirka**
Part of #46741

## What?

Update styles on TextareaControl to match the current [specs](https://www.figma.com/design/804HN2REV2iap2ytjRQ055/Core-System-Library?node-id=2518-56299). Namely, increased padding and leading.

## Testing Instructions

See Storybook for TextareaControl. See usages in the editor.

Note for code review: The [first commit](https://github.com/WordPress/gutenberg/pull/64586/commits/d2f3c0749f23080b2eec23c6ba72287b51c87f34) is simply a move of the styles in `utils/input/` which were only used for this TextareaControl component. The [effective changeset](https://github.com/WordPress/gutenberg/pull/64586/files/d2f3c0749f23080b2eec23c6ba72287b51c87f34..27388897ac283cb2d9a51b391143e4c0d1ae469f) is much smaller.

## Screenshots or screencast 

| Before | After |
|--------|-------|
|TextareaControl, before|TextareaControl, after|

How it looks in context with other controls:

| Before | After |
|--------|-------|
|Image block inspector, before|Image block inspector, after|

Or in this Additional CSS field:

| Before | After |
|--------|-------|
|Additional CSS input, before|Additional CSS input, after|

### CODE REVIEW

**Comment by mirka** [comment]
@WordPress/gutenberg-design The specs define a 10px vertical padding, but I tweaked it to 11.5px so the control height matches a standard InputControl when the TextareaControl is set to show one line of text only (set the `rows` prop to `1` to try this). This is probably a rare case, so we can also just use 10px if we don't care about it.

TextareaControl when rows prop is 1


Also, this upsizing on TextareaControl is unlikely to seriously break consumer usage, so I think we could get away with just shipping this change without first hiding it behind an opt-in prop. If there are no particular objections here, I'll go with that. (cc @WordPress/gutenberg-components)

**Comment by jasmussen** [comment]
Great improvement, thank you.

> The specs define a 10px vertical padding, but I tweaked it to 11.5px so the control height matches a standard InputControl when the TextareaControl is set to show one line of text only (set the rows prop to 1 to try this). This is probably a rare case, so we can also just use 10px if we don't care about it.

Just to be sure, which specs are these?

At a glance, this looks like a solid win for consistency, and that what exists in trunk is simly vestigial and overlooked. There's a lot of green in the code, so I'll defer to a code review for what's best there, but designwise, this looks great.

**Comment by mirka** [comment]
@jasmussen Thanks for confirming! I was referring to the specs [here](https://www.figma.com/design/e4tLacmlPuZV47l7901FEs/WordPress-Design-Library?node-id=28978-156668&t=l7QzzemxapTkAcW6-4).

**Comment by jasmussen** [comment]
Right, those definitely seem a bit out of date. I'll add the "Needs Figma Update" label here, but let's also make sure to get this right in [the new file](https://www.figma.com/design/804HN2REV2iap2ytjRQ055/System?node-id=2799-26256&t=Q6hHPfhgmV8GTH6P-1).

I think the only downside: 11.5 is so close to 12px which would be a nice round number. I guess it's the fact that most inputs are 40px, but the 13px font centered inside with its default lineheight that's dictating this odd number yes?

Thanks for diving in!

**Review by tyxla — COMMENTED** [review]
Love the utils cleanup here 👏 Makes so much sense to keep it all inline instead of in `/utils`.

**Comment by jameskoster** [comment]
It's not just about single-line textareas, but also ensuring the copy inside sits on the same baseline as regular text inputs, like this:

Screenshot 2024-08-19 at 13 50 33

I think it can be achieved by updating the following textarea properties:

* Padding: 10px 12px (we should update text inputs horizontal padding to be 12px too).
* Line-height: 20px (aligns with the proposal in https://github.com/WordPress/gutenberg/issues/64340).

In this case using a padding value off the 4px baseline seems unavoidable because the input height is 40, minus the line-height (20) equals 20, divide by 2 = 10.

**Comment by mirka** [comment]
Thank you for the design guidance! Here is the result, baseline matched with the other similar components:

Text input components side by side

I also wanted to mention a nuance with the Figma metrics. Not sure if this is fully consistent across the file, but at least for the one I'm looking at, the stroke is inset so the distance includes the border width:

Distance overlay in Figma

In code, it kind of depends on the implementation whether the border is "included" or not. In TextareaControl, the border is _not_ included so we need to subtract the border widths from the actual padding values. For InputControl-based components on the other hand, the border is rendered separately from the actual `input` element, so the padding values can include the border. This is something we'll need to be aware of when bringing over the metrics into code, as the off-by-one pixel can be noticeable in some contexts.

File: packages/components/src/textarea-control/styles/textarea-control-styles.ts
Comment (ciampo) [inline packages/components/src/textarea-control/styles/textarea-control-styles.ts]: Since we're trying to clarify the `9px` value, probably better to mention the `20px` line height too?

**Review by ciampo — APPROVED** [review]
Looking good from the code perspective

File: packages/components/src/textarea-control/styles/textarea-control-styles.ts
Comment (mirka) [reply packages/components/src/textarea-control/styles/textarea-control-styles.ts]: Good call, done 👍

**Comment by mirka** [comment]
Ok, I updated the PR description and screenshots. This is good to go on my end if we're happy with the changes 👍

**Comment by jameskoster** [comment]
> This is something we'll need to be aware of when bringing over the metrics into code, as the off-by-one pixel can be noticeable in some contexts.

Makes sense. 

Is there a reason not to include the spacing scale in `config-values.js`? I'm wondering if that might help simplify things, and reduce maintenance headaches. 

For similar components, code like `padding: calc( ${ CONFIG.gridUnit15 } - ${ CONFIG.borderWidth } )` might be easier to grok than hard-coded pixel values?

**Review by tyxla — APPROVED** [review]
LGTM 👍 🚀 

@jameskoster's suggestions on extracting some spacing vars make sense, but I think they can also be followed up with in another PR as needed.

**Comment by mirka** [comment]
I'll probably do some value abstracting when I update the input padding values from 16px to 12px (https://github.com/WordPress/gutenberg/pull/64520#discussion_r1722935668) 👍

**Pull request merged**
